core/src/main/java/org/zenoss/zep/impl/EventFanOutPlugin.java

```diff
@@ -42,14 +45,35 @@ public void setAmqpConnectionManager(AmqpConnectionManager amqpConnectionManager
     }
 
     @Override
+    public void startBatch(EventPostIndexContext context) throws Exception {
+        context.setPluginState(this, new ArrayList<EventSummary>(context.getIndexLimit()));
+    }
+
+    @Override
     public void processEvent(EventSummary eventSummary, EventPostIndexContext context) throws ZepException {
-        final String eventClass = eventSummary.getOccurrence(0).getEventClass();
-        try {
-            logger.debug("Publishing event to fan-out exchange: {}", eventSummary);
-            this.amqpConnectionManager.publish(this.exchangeConfiguration,
-                    ROUTING_KEY_PREFIX + sanitizeEventClass(eventClass), eventSummary);
-        } catch (AmqpException e) {
-            throw new ZepException(e);
+        List<EventSummary> events = (List<EventSummary>) context.getPluginState(this);
+        events.add(eventSummary);
+    }
+
+    @Override
+    public void endBatch(EventPostIndexContext context) throws Exception {
+        List<EventSummary> events = (List<EventSummary>) context.getPluginState(this);
+        AmqpException lastException = null;
+        int exceptionCount = 0;
+        for (EventSummary eventSummary : events) {
+            final String eventClass = eventSummary.getOccurrence(0).getEventClass();
+            try {
+                logger.debug("Publishing event to fan-out exchange: {}", eventSummary);
+                this.amqpConnectionManager.publish(this.exchangeConfiguration,
+                        ROUTING_KEY_PREFIX + sanitizeEventClass(eventClass), eventSummary);
+            } catch (AmqpException e) {
+                lastException = e;
+                exceptionCount++;
+            }
+        }
+        context.setPluginState(this, null);
+        if (exceptionCount > 0 && lastException != null) {
+            throw new ZepException(lastException);
```

Comment:
This exception will just get logged and swallowed. The last AMQP exception is likely to be identical to all the other AMQP exceptions encountered.